src/bmp_reader.cpp

```diff
@@ -36,43 +30,64 @@ bool ReadBMP(std::string imagepath, unsigned char *&header, unsigned char *&rgbD
     bitsPerPixel = *(unsigned short *)&header[BITS_PER_PIXEL_INDEX];
 
     if (headerSize > HEADER_SIZE)
-    {
-        // Header size is greater than 54 bytes, let's re-read the whole header.
-        delete header;
+    {  
+        delete[] header;
         header = new unsigned char[headerSize];
         rewind(file);
-        fread(header, 1, headerSize, file);
+        bytesRead = fread(header, 1, headerSize, file);
+        if (bytesRead != headerSize) {
+            std::cerr << "Error reading extended BMP header." << std::endl;
+            delete[] header;
+            fclose(file);
+            return false;
+        }
     }
     else if (headerSize == 0) // If header size was not specified within the file header.
-    {
-        // The BMP header has 54 bytes of that, the image should start right after the header.
+    {             
         headerSize = HEADER_SIZE;
     }
 
-    // Setting default values of imageSize if it was not found on the file header.
+   
     if (imageSize == 0)
     {
         imageSize = width * height * (bitsPerPixel / BITS_PER_BYTE);
     }
 
-    // Create a buffer
-    rgbData = new unsigned char[imageSize];
-
-    // Read the actual data from the file into the buffer.
-    fread(rgbData, 1, imageSize, file);
+   
+    rgbData = new unsigned char[imageSize];  
+    bytesRead = fread(rgbData, 1, imageSize, file);
+    if (bytesRead != imageSize) {
+        std::cerr << "Error reading BMP image data." << std::endl;
+        delete[] header;
+        delete[] rgbData;
+        fclose(file);
+        return false;
+    }
 
-    // Everything is in memory now, the file can be closed.
     fclose(file);
-
     return true;
 }
 
 bool WriteBMP(std::string imagepath, unsigned char *&header, unsigned char *&rgbData, size_t headerSize, size_t imageSize)
 {
     FILE *file;
-    file = fopen(imagepath.c_str(), "wb");
-    fwrite(header, 1, headerSize, file);
-    fwrite(rgbData, 1, imageSize, file);
+    file = fopen(imagepath.c_str(), "wb");  
```

Comment:
Why did you rewrite the WriteBMP function?

Reply:
Porque Replit no me estaba leyendo las bmp y decidí reecribir la función para ver si el editor me las ejecutaba